packages/terminal/src/browser/terminal-linkmatcher-files.ts

```diff
@@ -0,0 +1,176 @@
+                    return f !== undefined && f !== null && !f.isDirectory;
+                }
+            } catch (err) {
+                console.trace('Error validating ' + match);
+            }
+            return false;
+        };
+    }
+
+    getHandler(terminalWidget: TerminalWidgetImpl): (event: MouseEvent, link: string) => void {
+        return async (event, fullMatch) => {
+            const toOpen = await this.toURI(fullMatch, await terminalWidget.cwd);
+            if (!toOpen) {
+                return;
+            }
+            const position = await this.extractPosition(fullMatch);
+            let options = {};
+            if (position) {
+                options = {
+                    selection: {
+                        start: position
+                    }
+                };
+            }
+            try {
+                const opener = await this.openerService.getOpener(toOpen, options);
+                opener.open(toOpen, options);
+            } catch (err) {
+                console.error('Cannot open link ' + fullMatch, err);
+            }
+        };
+    }
+
+    protected async toURI(match: string, cwd: URI): Promise<URI | undefined> {
+        const path = await this.extractPath(match);
+        if (!path) {
+            return;
+        }
+        const pathObj = new Path(path);
+        return pathObj.isAbsolute ? cwd.withPath(path) : cwd.resolve(path);
+    }
+
+    protected async extractPosition(link: string): Promise<Position> {
+        const matches: string[] | null = (await this.getRegExp()).exec(link);
+        const info: Position = {
+            line: 1,
+            character: 1
+        };
+
+        if (!matches) {
+            return info;
+        }
+
+        const lineAndColumnMatchIndex = (await this.backendOs) === OS.Type.Windows ? winLineAndColumnMatchIndex : unixLineAndColumnMatchIndex;
+        for (let i = 0; i < lineAndColumnClause.length; i++) {
+            const lineMatchIndex = lineAndColumnMatchIndex + (lineAndColumnClauseGroupCount * i);
+            const rowNumber = matches[lineMatchIndex];
+            if (rowNumber) {
+                info.line = parseInt(rowNumber, 10) - 1;
+                const columnNumber = matches[lineMatchIndex + 2];
+                if (columnNumber) {
+                    info.character = parseInt(columnNumber, 10) - 1;
+                }
+                break;
+            }
+        }
+
+        return info;
+    }
+
+    protected async extractPath(link: string): Promise<string | undefined> {
+        const matches: string[] | null = (await this.getRegExp()).exec(link);
+        if (!matches) {
+            return undefined;
+        }
+        return matches[1];
+    }
+}
+
+// below reg exps are taken from
+// https://github.com/microsoft/vscode/blob/9edfff8d38d00bd87428af5b9dea7f41a7b5b4d9/src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts
```

Comment:
I think the correct link is: https://github.com/microsoft/vscode/blob/fbbc1aa80332189aa0d3006cb2159b79a9eba480/src/vs/workbench/contrib/terminal/browser/terminalLinkHandler.ts